src/main/java/org/folio/dcb/service/impl/BorrowingLibraryServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package org.folio.dcb.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.dcb.domain.dto.DcbTransaction;
+import org.folio.dcb.domain.dto.TransactionStatus;
+import org.folio.dcb.domain.dto.TransactionStatusResponse;
+import org.folio.dcb.domain.entity.TransactionEntity;
+import org.folio.dcb.repository.TransactionRepository;
+import org.folio.dcb.service.CirculationService;
+import org.folio.dcb.service.LibraryService;
+import org.springframework.stereotype.Service;
+
+import static org.folio.dcb.domain.dto.TransactionStatus.StatusEnum.AWAITING_PICKUP;
+import static org.folio.dcb.domain.dto.TransactionStatus.StatusEnum.OPEN;
+
+@Log4j2
+@Service("borrowingLibraryService")
+@RequiredArgsConstructor
+public class BorrowingLibraryServiceImpl implements LibraryService {
+
+  private final CirculationService circulationService;
+  private final TransactionRepository transactionRepository;
+
+  @Override
+  public TransactionStatusResponse createCirculation(String dcbTransactionId, DcbTransaction dcbTransaction, String pickupServicePointId) {
+    return null;
+  }
+
+  @Override
+  public void updateStatusByTransactionEntity(TransactionEntity transactionEntity) {
+
+  }
+
+  @Override
+  public void updateTransactionStatus(TransactionEntity dcbTransaction, TransactionStatus transactionStatus) {
+    log.debug("updateTransactionStatus:: Updating dcbTransaction {} to status {} ", dcbTransaction, transactionStatus);
+    var currentStatus = dcbTransaction.getStatus();
+    var requestedStatus = transactionStatus.getStatus();
+    if(OPEN == currentStatus && AWAITING_PICKUP == requestedStatus) {
+      circulationService.checkInByBarcode(dcbTransaction);
+      updateTransactionEntity(dcbTransaction, requestedStatus);
+    } else {
+      String error = String.format("updateTransactionStatus:: status update from %s to %s is not implemented", currentStatus, requestedStatus);
+      log.warn(error);
+      throw new IllegalArgumentException(error);
+    }
+  }
+
+  private void updateTransactionEntity(TransactionEntity transactionEntity, TransactionStatus.StatusEnum transactionStatusEnum) {
+    log.info("updateTransactionEntity:: updating transaction entity from {} to {}", transactionEntity.getStatus(), transactionStatusEnum);
```

Comment:
do we need this info at the info level ?
I believe the debug is more suitable.

@Vignesh-kalyanasundaram suggest please.

Reply:
Hi @DmytroBykov1 , I think if we change it to debug here as well, then we dont know what is happening during PUT API as the caller of this method is having logger level debug.